Realm/RLMManagedDictionary.mm

```diff
@@ -454,9 +483,9 @@ - (BOOL)isFrozen {
 - (instancetype)resolveInRealm:(RLMRealm *)realm {
     auto& parentInfo = _ownerInfo->resolve(realm);
     return translateErrors([&] {
-        return [[self.class alloc] initWithBackingCollection:_backingCollection.freeze(realm->_realm)
+        return [[RLMManagedDictionary alloc] initWithBackingCollection:_backingCollection.freeze(realm->_realm)
```

Comment:
Things which produce a copy of the current object use `self.class` rather than explicitly spelling out the class name  so that they work properly when called on subclasses. That's not directly relevant here, but it is the more generally correct pattern.